src/App/Fossa/Config/Container.hs

```diff
@@ -0,0 +1,343 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Config.Container (
+  mkSubCommand,
+  ImageText (..),
+  OutputFormat (..),
+  ContainerCommand,
+  ContainerScanConfig (..),
+  ContainerAnalyzeConfig (..),
+  ContainerTestConfig (..),
+  ContainerDumpScanConfig (..),
+  ContainerParseFileConfig (..),
+) where
+
+import App.Fossa.Config.Common (
+  CommonOpts (..),
+  ScanDestination (..),
+  collectAPIMetadata,
+  collectApiOpts,
+  collectRevisionOverride,
+  commonOpts,
+  defaultTimeoutDuration,
+  metadataOpts,
+  validateApiKey,
+ )
+import App.Fossa.Config.ConfigFile (
+  ConfigFile,
+  resolveConfigFile,
+ )
+import App.Fossa.Config.EnvironmentVars (EnvVars)
+import App.Fossa.Subcommand (EffStack, GetSeverity (getSeverity), SubCommand (SubCommand))
+import App.Types (
+  OverrideProject (OverrideProject),
+  ProjectMetadata,
+ )
+import Control.Effect.Diagnostics (Diagnostics, Validator, runValidation, validationBoundary)
+import Control.Effect.Lift (Has, Lift, sendIO)
+import Control.Timeout.Internal (Duration (Seconds))
+import Data.Flag (Flag, flagOpt, fromFlag)
+import Data.Text (Text)
+import Effect.Logger (Logger, Severity (SevDebug, SevInfo))
+import Effect.ReadFS (ReadFS)
+import Fossa.API.Types (ApiOpts (ApiOpts))
+import Options.Applicative (
+  CommandFields,
+  InfoMod,
+  Mod,
+  Parser,
+  argument,
+  auto,
+  command,
+  flag,
+  help,
+  hsubparser,
+  info,
+  internal,
+  long,
+  metavar,
+  option,
+  optional,
+  progDesc,
+  short,
+  str,
+  strOption,
+ )
+import Path (Abs, File, Path)
+import Path.IO (getCurrentDir, resolveFile)
+
+data OutputFormat
+  = TestOutputPretty
+  | TestOutputJson
+  deriving (Eq, Ord, Show)
+
+data NoUpload = NoUpload
+
+newtype ImageText = ImageText
+  { unImageText :: Text
+  }
+  deriving (Eq, Ord, Show)
+
+containerCmdInfo :: InfoMod a
+containerCmdInfo = progDesc "Run in container-scanning mode"
+
+mkSubCommand :: (ContainerScanConfig -> EffStack ()) -> SubCommand ContainerCommand ContainerScanConfig
+mkSubCommand = SubCommand "container" containerCmdInfo parser loadConfig mergeOpts
+
+mergeOpts ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  ) =>
+  Maybe ConfigFile ->
+  EnvVars ->
+  ContainerCommand ->
+  m ContainerScanConfig
+mergeOpts cfgfile envvars = \case
+  ContainerAnalyze opts -> AnalyzeCfg <$> mergeAnalyzeOpts cfgfile envvars opts
+  ContainerTest opts -> TestCfg <$> mergeTestOpts cfgfile envvars opts
+  ContainerParseFile fp -> ParseCfg <$> mergeParseFileOpts cfgfile envvars fp
+  ContainerDumpScan opts -> DumpCfg <$> mergeDumpOpts cfgfile envvars opts
```

Comment:
Not sure I agree here (or rather, not sure **yet**, since I haven't tried it).  These are small enough that combining them into one module is significantly easier than splitting them up.  I can take a swing at this and see if I like it, but my initial reaction is that it might be more confusing by being too spread out.